### PR TITLE
Feature/issue 360 data paths currency

### PR DESCRIPTION
**Pull request by jgimeno**
Now it uses the currency as datapath.

Closes #
Last checkbox in https://github.com/kowala-tech/kcoin/issues/360

### CODE REVIEW

**Comment by rgeraldes** [comment]
This PR should not be merged right away - requires multiple reviewers.

File: cmd/kcoin/chaincmd.go [hunk 1]
```diff
@@ -157,8 +159,14 @@ func initGenesis(ctx *cli.Context) error {
 }
 
 //extractKeyCoin returns the keycoin used for launching the client. For now it always return kusd.
```
Comment (acroca) [inline cmd/kcoin/chaincmd.go]: Update the comment.

File: cmd/kcoin/chaincmd.go [hunk 2]
```diff
@@ -157,8 +159,14 @@ func initGenesis(ctx *cli.Context) error {
...
-func extractKeyCoin(context *cli.Context) string {
-	return "kusd"
+func extractKeyCoin(ctx *cli.Context) string {
+	kcoin := ctx.GlobalString(utils.CurrencyFlag.Name)
+
+	if kcoin == "" {
```
Comment (acroca) [inline cmd/kcoin/chaincmd.go]: Its default value is `"kusd"`. Why do we need this if? Just for safety?

**Comment by acroca** [comment]
Do we want to limit the values we accept for the currency or it's fine to keep it open? I don't see any harm in running it with an invalid currency, but perhaps someone will use `kUSD` by mistake instead of `kusd`...

File: cmd/kcoin/chaincmd.go (hunk 2 above)
Comment (jgimeno) [reply cmd/kcoin/chaincmd.go]: Yeah, in theory never will happen.

**Comment by jgimeno** [comment]
@acroca for now it fails with an invalid currency, since it checks for the available ones when generating the genesis.

**Comment by jgimeno** [comment]
Even kUSD

**Comment by acroca** [comment]
awesome. Does it show a meaningful error message? :)

**Comment by jgimeno** [comment]
Absolutely @acroca ;P 

It says invalid kcoin.

**Pull request merged**